### PR TITLE
Add API for managing product association types

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | kind of |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets | - |
| License         | MIT |

API endpoints for product association types were missing.

### CODE REVIEW

File: tests/Controller/ProductAssociationTypeApiTest.php [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    /**
+     * @param ProductAssociationTypeInterface $productAssociationType
+     *
+     * @return string
+     */
+    private function getAssociationTypeUrl(ProductAssociationTypeInterface $productAssociationType)
+    {
+        return 'api/v1/product-association-types/' . $productAssociationType->getCode();
+    }
+
```
Comment (lchrusciel) [inline tests/Controller/ProductAssociationTypeApiTest.php]: Redundant blank line

File: tests/DataFixtures/ORM/resources/product_association_types.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+Sylius\Component\Product\Model\ProductAssociationTypeTranslation:
+    associationTypeTranslation1pl:
```
Comment (lchrusciel) [inline tests/DataFixtures/ORM/resources/product_association_types.yml]: We are really inconsistent in terms of defining references in fixtures... Something should be done with it.

File: tests/Responses/Expected/product_association_type/create_response.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+    "id": @integer@,
+    "code": "cross_sell",
+    "createdAt": "@string@.isDateTime()",
+    "updatedAt": "@string@.isDateTime()",
+    "translations": {
+        "en_US": {
+            "locale": "en_US",
+            "id": @integer@,
```
Comment (lchrusciel) [inline tests/Responses/Expected/product_association_type/create_response.json]: It is a first key in array usually.

File: tests/Responses/Expected/product_association_type/update_validation_fail_response.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                    "en_US": {
+                        "children": {
+                            "name": {
+                                "errors": [
+                                    "Please enter association type name."
+                                ]
+                            }
+                        }
+                    },
+                    "nl_NL": {
```
Comment (lchrusciel) [inline tests/Responses/Expected/product_association_type/update_validation_fail_response.json]: We should make theses keys relevant to provided translations(`pl`)

File: src/Sylius/Bundle/ProductBundle/Resources/config/serializer/Model.ProductAssociationType.yml [hunk 5]
```diff
@@ -15,6 +15,18 @@ Sylius\Component\Product\Model\ProductAssociationType:
             expose: true
             type: array
             groups: [Default, Detailed]
-    virtual_properties:
-        getName:
-            serialized_name: name
+        createdAt:
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ProductBundle/Resources/config/serializer/Model.ProductAssociationType.yml]: I just have checked a failing build. Becuase you have updated object serialization, response for product created with association should also be updated. We are missing `createdAt` and `updatedAt` there.

File: tests/Responses/Expected/product_association_type/update_validation_fail_response.json (hunk 4 above)
Comment (pamil) [inline tests/Responses/Expected/product_association_type/update_validation_fail_response.json]: Actually switched from `pl` to `nl_NL` :)

File: src/Sylius/Bundle/ProductBundle/Resources/config/serializer/Model.ProductAssociationType.yml (hunk 5 above)
Comment (pamil) [inline src/Sylius/Bundle/ProductBundle/Resources/config/serializer/Model.ProductAssociationType.yml]: Done! 🎉

**Comment by lchrusciel** [comment]
Thanks Kamil!

**Pull request merged**